datatransfer/impl/graphsync/graphsync_impl_test.go

```diff
@@ -873,6 +874,64 @@ func TestRespondingToPushGraphsyncRequests(t *testing.T) {
 	})
 }
 
+func TestResponseHookWhenExtensionNotFound(t *testing.T) {
+	// create network
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+	gsData := newGraphsyncTestingData(ctx, t)
+	host1 := gsData.host1 // initiator and data sender
+	host2 := gsData.host2 // data recipient, makes graphsync request for data
+	voucher := fakeDTType{"applesauce"}
+	link := gsData.loadUnixFSFile(t, false)
+
+	// setup receiving peer to just record message coming in
+	dtnet2 := network.NewFromLibp2pHost(host2)
+	r := &receiver{
+		messageReceived: make(chan receivedMessage),
+	}
+	dtnet2.SetDelegate(r)
+
+	gsr := &fakeGraphSyncReceiver{
+		receivedMessages: make(chan receivedGraphSyncMessage),
+	}
+	gsData.gsNet2.SetDelegate(gsr)
+
+	gs1 := gsData.setupGraphsyncHost1()
+	dt1 := NewGraphSyncDataTransfer(ctx, host1, gs1)
+
+	t.Run("when it's not our extension, does not error and does not validate", func(t *testing.T) {
+		//register a hook that validates the request so we don't fail in gs because the request
+		//never gets processed
+		validateHook := func(p peer.ID, req graphsync.RequestData, ha graphsync.RequestReceivedHookActions) {
```

Comment:
in particular here, want to make sure my understanding is correct

Reply:
your understanding is correct for this case. gsData.allSelector is a "get the whole dag" selector which is not accepted by default. Selectors that are less open ended are accepted and processed in graphsync by default.

Sorry if that's a subtle distinction. (The specific reason is the all selector will fail is that it contains a recursive selection with no limit -- the default policy is recursive selections must have a limit, and that limit must be < 100)